### PR TITLE
sis-scraper: fix saturday class failure

### PR DESCRIPTION
**Pull request by johnnyapol**
BIOL-4720: MOLECULAR BIOLOGY LAB has a saturday meeting time. This change causes indexing errors

### CODE REVIEW

**Review by benjaminrsherman — APPROVED** [review]

File: sis_scraper/main.py [hunk 1]
```diff
@@ -286,7 +287,7 @@ def toTitle(text):
             for section in course["sections"]:
                 crn_to_courses[section["crn"]] = course["id"]
 
-                conflict = [0] * (64 * 3)
+                conflict = [0] * (64 * 4)
                 for time in section["timeslots"]:
                     for day in time["days"]:
                         for i in range(500, 2400, 100):
```
Comment (elihschiff) [inline sis_scraper/main.py]: It wont let me make a comment on the line, but line 314 is now also wrong. I think it needs to be `f"\t{crn}u32 => [{int(conflict[:64], 2)}, {int(conflict[64:128], 2)}, {int(conflict[128:192], 2)}, {int(conflict[192:], 2)}],\n"`

**Review by elihschiff — CHANGES_REQUESTED** [review]

File: sis_scraper/main.py (hunk 1 above)
Comment (benjaminrsherman) [reply sis_scraper/main.py]: That will cause issues with quacs-rs which expects only three integers there.  For example, line 308 will need to be updated to `pub static CRN_TIMES: Map<u32, [u64; 4]> = phf_map! {`.

Comment (benjaminrsherman) [reply sis_scraper/main.py]: The conflicts goes from 5am to midnight daily.  If we cut that down to 7am to 11pm, we'll be able to exactly fit 6 days of conflicts in 3 64 bit numbers (unless I messed up the math and it's short by a bit or two).

Comment (benjaminrsherman) [reply sis_scraper/main.py]: (That would be changing this line from 500 and 2400 as the bounds to 700 and 2300)

File: sis_scraper/main.py [hunk 2]
```diff
@@ -286,7 +287,7 @@ def toTitle(text):
             for section in course["sections"]:
                 crn_to_courses[section["crn"]] = course["id"]
 
-                conflict = [0] * (64 * 3)
+                conflict = [0] * (64 * 4)
```
Comment (elihschiff) [inline sis_scraper/main.py]:
```suggestion
                conflict = [0] * (64 * 3)
```

File: sis_scraper/main.py [hunk 3]
```diff
@@ -277,6 +277,7 @@ def toTitle(text):
         "W": 2 * 19 * 2,
         "R": 3 * 19 * 2,
         "F": 4 * 19 * 2,
+        'S': 5 * 19 * 2,
```
Comment (elihschiff) [inline sis_scraper/main.py]:
```suggestion
        'S': 5 * 16 * 2,
```

File: sis_scraper/main.py (hunk 1 above)
Comment (elihschiff) [inline sis_scraper/main.py]:
```suggestion
                        for i in range(700, 2300, 100):
```

File: sis_scraper/main.py [hunk 4]
```diff
@@ -277,6 +277,7 @@ def toTitle(text):
         "W": 2 * 19 * 2,
         "R": 3 * 19 * 2,
         "F": 4 * 19 * 2,
```
Comment (elihschiff) [inline sis_scraper/main.py]:
```suggestion
        "F": 4 * 16 * 2,
```

File: sis_scraper/main.py [hunk 5]
```diff
@@ -277,6 +277,7 @@ def toTitle(text):
         "W": 2 * 19 * 2,
         "R": 3 * 19 * 2,
```
Comment (elihschiff) [inline sis_scraper/main.py]:
```suggestion
        "R": 3 * 16 * 2,
```

File: sis_scraper/main.py [hunk 6]
```diff
@@ -277,6 +277,7 @@ def toTitle(text):
         "W": 2 * 19 * 2,
```
Comment (elihschiff) [inline sis_scraper/main.py]:
```suggestion
        "W": 2 * 16 * 2,
```

Comment (elihschiff) [inline sis_scraper/main.py]: Please also fix the 2 lines above to say 16, github wont let me comment on them

File: sis_scraper/main.py (hunk 1 above)
Comment (elihschiff) [reply sis_scraper/main.py:293]: This no longer needs to be done

**Review by elihschiff — CHANGES_REQUESTED** [review]
I left some changes that should fix it to work with the new timeframe of 7am-11pm

**Comment by johnnyapol** [comment]
@elihschiff the commit I just pushed should address your changes.

**Review by elihschiff — APPROVED** [review]
This looks good to me. I am too lazy to run it but worst case if it crashes we can fix it later :man_shrugging:

**Pull request merged**